nengo/cache.py

```diff
@@ -321,6 +321,11 @@ def shrink(self, limit=None):
             Maximum size of the cache in bytes.
         """
         if self.readonly:
+            warnings.warn("Cannot shrink a readonly cache.")
```

Comment:
This might give more warnings than we want. There are basically two situations where the cache might be readonly:

1. [The lock cannot be acquired.](https://github.com/nengo/nengo/blob/79fbe66ca6304b5a4294103a63e878a4e210de24/nengo/cache.py#L200) In that we already give a warning.
2. The user explicitly configures the cache to be readonly. In that case the simulator (or builder with this PR I assume) will still call shrink and the user cannot deactivate this call (and doesn't need to), but will get an irrelevant warning.

However, it might be a useful warning for developers when debugging?